src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md

```diff
@@ -0,0 +1,712 @@
+
+See the [arrival rate](#arrival-rate) section for details.
+
+#### Options
+
+| Option            | Type    | Description                                                                             | Default |
+|-------------------|---------|-----------------------------------------------------------------------------------------|---------|
+| `startRate`       | integer | Number of iterations to execute each `timeUnit` period at test start.                   | `0`     |
+| `timeUnit`        | string  | Period of time to apply the `startRate` the `stages` `target` value.                    | `"1s"`  |
+| `stages`          | array   | Array of objects that specify the target number of iterations to ramp up or down to.    | `[]`    |
+| `preAllocatedVUs` | integer | Number of VUs to pre-allocate before test start in order to preserve runtime resources. | -       |
+| `maxVUs`          | integer | Maximum number of VUs to allow during the test run.                                     | -       |
+
+#### When to use
+
+If you need your tests to not be affected by the system-under-test's performance, and
+would like to ramp the number of iterations up or down during specific periods of time.
+
+#### Examples
+
+- Execute a variable RPS test, starting at 50, ramping up to 200 and then back to 0,
+  over a 1 minute period:
+
+<div class="code-group" data-props='{"labels": [ "ramping-arr-rate.js" ], "lineNumbers": "[true]"}'>
+
+```js
+import http from 'k6/http';
+
+export let options = {
+  discardResponseBodies: true,
+  scenarios: {
+    contacts: {
+      executor: 'ramping-arrival-rate',
+      startRate: 50,
+      timeUnit: '1s',
+      preAllocatedVUs: 50,
+      maxVUs: 100,
+      stages: [
+        { target: 200, duration: '30s' },
+        { target: 0, duration: '30s' },
+      ],
+    }
+  }
+};
+
+export default function() {
+  http.get('https://test.k6.io/contacts.php');
+}
+```
+
+</div>
+
+
+### Externally controlled
+
+Control and scale execution at runtime via [k6's REST API](/misc/k6-rest-api) or
+the [CLI](https://k6.io/blog/how-to-control-a-live-k6-test).
+
+This executor formalizes what was previously possible globally, and using it is
+required in order to use the `pause`, `resume`, and `scale` CLI commands. Also note
+that arguments to `scale` to change the amount of active or maximum VUs only affect
+the externally controlled executor.
+
+#### Options
+
+| Option     | Type    | Description                                         | Default |
+|------------|---------|-----------------------------------------------------|---------|
+| `vus`      | integer | Number of VUs to run concurrently.                  | -       |
+| `maxVUs`   | integer | Maximum number of VUs to allow during the test run. | -       |
+| `duration` | string  | Total test duration.                                | -       |
+
+#### When to use
+
+If you want to control the number of VUs while the test is running.
```

Comment:
```suggestion
If you want to control the number of VUs while the test is running.

Important: this is the only executor that is not supported in `k6 cloud`, it can only be used locally with `k6 run`.
```